### PR TITLE
add a fix to upgrade playbook in case of customer change the configuration

### PR DESCRIPTION
**Pull request by mgold1234**
the configuration file.
related to https://issues.redhat.com/browse/THEEDGE-3223?focusedId=21963643&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-21963643

# Description

FIXES: 

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

### CODE REVIEW

**Comment by loadtheaccumulator** [comment]
What happens if a line other than proxy is added?

**Comment by de1987** [comment]
> What happens if a line other than proxy is added?

agree with Jonathan
and if it's empty? or null?

Also, what is the security it could imply?

**Review by acosferreira — CHANGES_REQUESTED** [review]
validates if has all the required keys present 
and should just change the contenturl value to the updated value.
this will prevent to change the file in case the user add any other confi besides the proxy

**Comment by loadtheaccumulator** [comment]
> ... should just change the contenturl value to the updated value. this will prevent to change the file in case the user add any other confi besides the proxy

+1 

@mgold1234 That is what I was recommending in our chat. Any values passed via the playbook should be added/modified instead of overwriting the entire file with customizations read from local and added back.

File: templates/template_playbook_dispatcher_ostree_rebase_payload_prod.test.yml [hunk 1]
```diff
@@ -11,41 +11,42 @@
     os_tree_ref: "rhel/9/x86_64/edge"
     ostree_gpg_verify: "false"
     ostree_gpg_keypath: "/etc/pki/rpm-gpg/"
-    ostree_remote_template: |
-      [remote "{{ ostree_remote_name }}"]
```
Comment (ldjebran) [inline templates/template_playbook_dispatcher_ostree_rebase_payload_prod.test.yml:15]: I think we are missing the config section

**Review by ldjebran — COMMENTED** [review]
awesome resolution, but we should be also able  to create the file if it's missing 
and of course with the config section.
```
[remote “rhel-edge”]


```

File: templates/template_playbook_dispatcher_ostree_rebase_payload_prod.test.yml (hunk 1 above)
Comment (ldjebran) [reply templates/template_playbook_dispatcher_ostree_rebase_payload_prod.test.yml:15]: is "ostree_remote_name" var is a constant or may change over updates ?

Comment (ldjebran) [reply templates/template_playbook_dispatcher_ostree_rebase_payload_prod.test.yml:15]: I think we should
 what  do you think @loadtheaccumulator @acosferreira

Comment (loadtheaccumulator) [reply templates/template_playbook_dispatcher_ostree_rebase_payload_prod.test.yml:15]: @mgold1234 @ldjebran I think if file exists would be a good check. You would do it programmatically out of habit, so probably a good practice to continue here.

Comment (loadtheaccumulator) [reply templates/template_playbook_dispatcher_ostree_rebase_payload_prod.test.yml:15]: Take a look at backup and create options.

File: templates/template_playbook_dispatcher_ostree_upgrade_payload.yml [hunk 2]
```diff
@@ -11,41 +11,42 @@
...
+      dldVWmFVa2RsTURSek0zRkJUMnB0YzNWeldYWndTVWt4VHk5WGRXZFhkbVlyUldndlIxSTVjRkZv
+      VlhCTVdFNUhTVmRMWlVGamJEY0tZVXhOTTNneU1YTjNTWFpLYTFwcGNGTnlabFJaYUdkRVFVUTVW
+      MVV6ZEM5UVpFVTJOM0EyWTNnNVFrRnFUWEF4YUZsV1VETlNUMk5NUXpORUswdFlNd3B6WW14T05X
+      eGxXSEJQYXowS1BYRkJWVlVLTFMwdExTMUZUa1FnVUVkUUlGTkpSMDVCVkZWU1JTMHRMUzB0Q2c9
       PQ==
   tasks:
     - name: apply templated ostree remote config
-      ansible.builtin.copy:
-        content: "{{ ostree_remote_template }}"
+      ansible.builtin.lineinfile:
```
Comment (loadtheaccumulator) [inline templates/template_playbook_dispatcher_ostree_upgrade_payload.yml:38]: Yeah. Please backup the file before modifying, and take a look at the create option to create the file if it doesn't exist.

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]

**Comment by loadtheaccumulator** [comment]
https://github.com/RedHatInsights/edge-api/pull/1994 replaces this PR.

**Pull request closed without merging**